Integrations/integration-Remedy_AR.yml

```diff
@@ -0,0 +1,162 @@
+    import json
+    import urllib
+    import os
+    # disable insecure warnings
+    requests.packages.urllib3.disable_warnings()
+
+    if not demisto.params()['proxy']:
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    ''' HELPER FUNCTIONS '''
+    def http_request(method, url_suffix, data, headers):
+        data = {} if data is None else data
+        LOG('running request with url=%s\tdata=%s\theaders=%s' % (BASE_URL + url_suffix,
+            data, headers))
+        try:
+            res = requests.request(method,
+                BASE_URL + url_suffix,
+                verify=USE_SSL,
+                data=data,
+                headers=headers
+            )
+            if res.status_code not in (200, 204):
+                raise Exception('Your request failed with the following error: ' + res.reason)
+        except Exception, e:
+            raise
+        return res
+
+    ''' GLOBAL VARS '''
+    SERVER_URL=demisto.params()['server']
+    USERNAME=demisto.params()['credentials']['identifier']
+    PASSWORD=demisto.params()['credentials']['password']
+    BASE_URL=SERVER_URL + '/api'
+    USE_SSL = not demisto.params().get('insecure', False)
+    DEFAULT_HEADERS = {
+        'Content-Type': 'application/json'
+    }
+
+    ''' FUNCTIONS '''
+    def login():
+        cmd_url = '/jwt/login'
+        data = {
+            'username': USERNAME,
+            'password': PASSWORD
+        }
+        login_headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+        result = http_request('POST', cmd_url, data, login_headers)
+        return result
+
+    def logout():
+        cmd_url = '/jwt/logout'
+        http_request('POST', cmd_url, None, DEFAULT_HEADERS)
+
+    def get_server_details(qualification, fields):
+        # URL Encodes qualification
+        qualification = urllib.quote_plus(qualification)
+        # Adds fields to filter by
+        fields = 'fields=values(' + fields + ')'
+        cmd_url = '/arsys/v1/entry/AST:ComputerSystem/?q=' + qualification + '&' + fields
+        result = http_request('GET', cmd_url, None, DEFAULT_HEADERS).json()
+        server_details = result['entries'][0]['values']
+        context = {
+            'Remedy': {
+                'ServerDetails': server_details
+            }
+        }
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': json.dumps(server_details),
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Remedy AR Server Details', server_details),
+            'EntryContext': context
+        }
+        return entry
+
+    ''' EXECUTION CODE '''
+    auth = login()
+    token = auth.content
+    DEFAULT_HEADERS['Authorization'] = 'AR-JWT ' + token
+
+    LOG('command is %s' % (demisto.command(), ))
```

Comment:
do we want this log? I think it can be removed.

Reply:
It's good to have, cause in case of error, we'll be able to know in which command it occured. We are using it in all recent Python integrations.